src/brsTypes/components/RoXMLElement.ts

```diff
@@ -0,0 +1,98 @@
+import { BrsBoolean, BrsString, BrsValue, ValueKind } from "../BrsType";
+import { BrsType } from "..";
+import { Callable, StdlibArgument } from "../Callable";
+import { Interpreter } from "../../interpreter";
+import { XmlDocument, XmlElement } from "xmldoc";
+
+export class RoXMLElement extends BrsComponent implements BrsValue {
+    readonly kind = ValueKind.Object;
+    private xmlNode?: any;
+
+    constructor() {
+        super("roXMLElement");
+
+        this.registerMethods({
+            ifXMLEelement: [this.parse, this.getName, this.getNamedElementsCi, this.getAttributes],
+        });
+    }
+
+    public setXMLElem(xmlElem: XmlElement) {
+        this.xmlNode = xmlElem;
+        return this;
+    }
+
+    toString(parent?: BrsType) {
+        return "<Component: roXMLElement>";
+    }
+
+    equalTo(other: BrsType) {
+        return BrsBoolean.False;
+    }
+
+    private parse = new Callable("parse", {
+        signature: {
+            args: [new StdlibArgument("str", ValueKind.String)],
+            returns: ValueKind.Boolean,
+        },
+        impl: (interpreter: Interpreter, str: BrsString) => {
+            try {
+                this.xmlNode = new XmlDocument(str.value);
+                return BrsBoolean.from(true);
+            } catch (err) {
+                this.xmlNode = undefined;
+                return BrsBoolean.from(false);
+            }
+        },
+    });
+
+    private getName = new Callable("getName", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (interpreter: Interpreter, str: BrsString) => {
+            return new BrsString(this.xmlNode?.name ?? "");
+        },
+    });
+
+    private getNamedElementsCi = new Callable("getNamedElementsCi", {
+        signature: {
+            args: [new StdlibArgument("str", ValueKind.String)],
+            returns: ValueKind.Object,
+        },
+        impl: (interpreter: Interpreter, str: BrsString) => {
+            let neededKey = str.value.toLowerCase();
+            let arr = [];
+            for (let item of this.xmlNode?.children ?? []) {
+                if (item.name?.toLowerCase() === neededKey) {
+                    arr.push(new RoXMLElement().setXMLElem(item));
+                }
+            }
+            return new RoArray(arr);
+        },
+    });
+
+    private getAttributes = new Callable("getAttributes", {
+        signature: {
+            args: [],
+            returns: ValueKind.Object,
+        },
+        impl: (interpreter: Interpreter, str: BrsString) => {
+            return this.convertObjectToBrsAA(this.xmlNode?.attr ?? {});
+        },
+    });
+
+    private convertObjectToBrsAA(object_: any) {
+        let array = [];
+        for (let key in object_) {
+            array.push({
+                name: new BrsString(key),
+                value: new BrsString(object_[key]),
+            });
+        }
+        return new RoAssociativeArray(array);
```

Comment:
got this error:
![image](https://user-images.githubusercontent.com/33808928/104211719-18c16d80-543d-11eb-90b9-0379f1794c1c.png)

Reply:
Ah, that's because `thix.xmlNode` is of type `any` at the top of this file, so `this.xmlNode?.attr` also has type `any`.  TypeScript is only able to guarantee that `Object.entries(attrs)` returns an array of `[ string, unknown ]` pairs, because there's no restriction on `attrs`.  The simplest way to fix that is to provide proper typing for `this.xmlNode`, which lets the compiler take care of the rest.  This diff should take care of that:

```diff
diff --git a/src/brsTypes/components/RoXMLElement.ts b/src/brsTypes/components/RoXMLElement.ts
index 8753ed0..9fd58d9 100644
--- a/src/brsTypes/components/RoXMLElement.ts
+++ b/src/brsTypes/components/RoXMLElement.ts
@@ -9,7 +9,7 @@ import { XmlDocument, XmlElement } from "xmldoc";
 
 export class RoXMLElement extends BrsComponent implements BrsValue {
     readonly kind = ValueKind.Object;
-    private xmlNode: any;
+    private xmlNode?: XmlElement;
 
     constructor() {
         super("roXMLElement");
@@ -62,7 +62,7 @@ export class RoXMLElement extends BrsComponent implements BrsValue {
             let neededKey = str.value.toLowerCase();
             let arr = [];
             for (let item of this.xmlNode?.children ?? []) {
-                if (item.name?.toLowerCase() === neededKey) {
+                if (item.type === "element" && item.name?.toLowerCase() === neededKey) {
                     let childXmlElement = new RoXMLElement();
                     childXmlElement.xmlNode = item;
                     arr.push(childXmlElement);
@@ -78,15 +78,15 @@ export class RoXMLElement extends BrsComponent implements BrsValue {
             returns: ValueKind.Object,
         },
         impl: (interpreter: Interpreter, str: BrsString) => {
-            let array = [];
             let attrs = this.xmlNode?.attr ?? {};
-            for (let key in attrs) {
-                array.push({
-                    name: new BrsString(key),
-                    value: new BrsString(attrs[key]),
-                });
-            }
-            return new RoAssociativeArray(array);
+            return new RoAssociativeArray(
+                Object.entries(attrs).map(
+                    ([name, value]) => ({
+                        name: new BrsString(name),
+                        value: new BrsString(value)
+                    })
+                )
+            );
         },
     });
 }
```

Reply:
Thanks for help and git diff